seed/utils/filter_state.py

```diff
@@ -0,0 +1,175 @@
+"""
+:copyright (c) 2014 - 2022, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.db.utils import DataError
+from django.http import JsonResponse, HttpResponse
+from rest_framework import status
+from rest_framework.request import Request
+from seed.lib.superperms.orgs.models import Organization
+from seed.models import (VIEW_LIST, VIEW_LIST_PROPERTY, VIEW_LIST_TAXLOT,
+                         Column, ColumnListProfile, ColumnListProfileColumn,
+                         Cycle, PropertyView)
+from seed.models import TaxLotProperty, TaxLotView
+from seed.search import build_view_filters_and_sorts, FilterException
+from seed.serializers.pint import (apply_display_unit_preferences)
+from typing import Optional, Literal
+
+
+def get_filtered_results(request: Request, inventory_type: Literal['property', 'taxlot'], profile_id: int) -> HttpResponse:
+    page = request.query_params.get('page', 1)
+    per_page = request.query_params.get('per_page', 1)
+    org_id = request.query_params.get('organization_id')
+    cycle_id = request.query_params.get('cycle')
+    # check if there is a query paramater for the profile_id. If so, then use that one
+    profile_id = request.query_params.get('profile_id', profile_id)
+
+    if not org_id:
+        return JsonResponse(
+            {'status': 'error', 'message': 'Need to pass organization_id as query parameter'},
+            status=status.HTTP_400_BAD_REQUEST)
+    org = Organization.objects.get(id=org_id)
+
+    if cycle_id:
+        cycle = Cycle.objects.get(organization_id=org_id, pk=cycle_id)
+    else:
+        cycle = Cycle.objects.filter(organization_id=org_id).order_by('name')
+        if cycle:
+            cycle = cycle.first()
+        else:
```

Comment:
This might be small but would we benefit from changing 'else' to 'if not' and bumping it back an indent?  Then we could check both if finding a cycle and if looking up a cycle by id failed.

Reply:
This makes sense to me